lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -615,17 +615,17 @@ namespace Js
                         recycler->ReportExternalMemoryFailure(length);
                     }
                 }
-                else
-                {
-                    JavascriptError::ThrowOutOfMemoryError(GetScriptContext());
-                }
             }
 
             if (buffer != nullptr)
             {
                 bufferLength = length;
                 ZeroMemory(buffer, bufferLength);
             }
+            else
```

Comment:
LGTM - although Curtis can answer history behind why we didn't have this in the first place?